run_operations.py

```diff
@@ -1,394 +1,294 @@
+
+    return datasets
+
+def get_newest_data_date(template_file):
+
+    dset_template = dataset_template.Template(template_file)
+    ssaraopt_string = dset_template.generate_ssaraopt_string()
+
+    ssaraopt_cmd = 'ssara_federated_query.py {} --print'.format(ssaraopt_string)
+
+    ssara_output = subprocess.check_output(ssaraopt_cmd, shell=True)
+
+    newest_data = ssara_output.decode('utf-8').split("\n")[-2]
+
+    return datetime.strptime(newest_data.split(",")[3], DATE_FORMAT)
+
+def get_last_downloaded_date(dset):
+
+    dataset_line = None
+    with open(STORED_DATE_FILE, 'r') as date_file:
+        for line in date_file.readlines():
+            if dset in line:
+                dataset_line = line
+                break
+
+    if dataset_line is not None:
+        last_date = dataset_line.split(": ")[1].strip("\n")
+    else:
+        last_date = datetime.strftime(datetime(1970, 1, 1, 0, 0, 0), DATE_FORMAT)
+    
+    return datetime.strptime(last_date, DATE_FORMAT)
+
+def run_process_rsmas(inps, template_file, dataset):
+
+    process_rsmas_options = []
+
+    if inps.startssara:
+        process_rsmas_options.append('--startssara')
+    if inps.stopssara:
+        process_rsmas_options.append('--stopssara')
+    if inps.startprocess:
+        process_rsmas_options.append('--startprocess')
+    if inps.stopprocess:
+        process_rsmas_options.append('--stopprocess')
+    if inps.startpysar:
+        process_rsmas_options.append('--startpysar')
+    if inps.stoppysarload:
+        process_rsmas_options.append('--stoppysarload')
+    if inps.stoppysar:
+        process_rsmas_options.append('--stoppysar')
+    if inps.startinsarmaps:
+        process_rsmas_options.append('--startinsarmaps')
+
+    if len(process_rsmas_options) == 0:
+        process_rsmas_options.append('--insarmaps')
+
+    process_rsmas_options = ' '.join(process_rsmas_options)
+
+    process_rsmas_cmd = "process_rsmas.py {} {} --submit".format(template_file, process_rsmas_options)
+
+    process_rsmas = subprocess.check_output(process_rsmas_cmd, shell=True).decode('utf-8')
+    
+    job_number = process_rsmas.split('\n')[0]
+
+    file_base = os.path.join(SCRATCH_DIRECTORY, dataset)
+
+    stdout_file = os.path.join(file_base, "/process_rsmas_{}.o".format(job_number))
+    stderr_file = os.path.join(file_base, "/process_rsmas_{}.e".format(job_number))
 
+    logger_run_operations.log(loglevel.INFO, "Job Number: {}".format(job_number))
+    logger_run_operations.log(loglevel.INFO, "Output Files: {}, {}".format(stdout_file, stderr_file))
+
+    return [stdout_file, stderr_file], job_number
+
+def copy_output_file(output_file, dataset):
+    
+    if os.path.exists(output_file) and os.path.isfile(output_file):
+
+        base = os.path.join(OPERATIONS_DIRECTORY, 'ERRORS', dataset, '')
+
+        if not os.path.exists(base):
+            os.makedirs(base)
+
+        destination = base + str(datetime.now().strftime("%m-%d-%Y")) + os.path.splitext(output_file)[1]
```

Comment:
It could be cleaner to do a `os.path.join()` to create `destination`. That would also mean you could remove the empty string in L183, which seems hacky